src/models.ts

```diff
@@ -561,3 +576,164 @@ export interface MemberTeamSummaryRep {
 	 */
 	name: string;
 }
+
+export interface IFlagStore {
+	storeUpdates: EventEmitter<boolean | null>;
+	// We fire a storeReady event because this will always exist compared to 'ready' listener on LDClient
+	// which may be reinitialized
+	storeReady: EventEmitter<boolean | null>;
+	ready: EventEmitter<boolean | null>;
+	reload(): Promise<void>;
+	start(): Promise<void>;
+	on(event: string, cb: (keys: string) => void): void;
+	removeAllListeners(): Promise<void>;
+	stop(): Promise<void>;
+	getFeatureFlag(flag: string, fullFlag?: boolean): Promise<FlagWithConfiguration>;
+	forceFeatureFlagUpdate(flagKey: string): Promise<void>;
+	allFlags(): Promise<FlagConfiguration[] | LDFeatureStoreKindData>;
+	getFlagConfig(flag: string): Promise<FlagConfiguration>;
+	getFlagMetadata(flag: string): Promise<FeatureFlag>;
+	allFlagsMetadata(): Promise<Dictionary<FeatureFlag>>;
+	listFlags(): Promise<Array<string>>;
+	executeAndUpdateFlagStore(
+		func: (
+			projectKey: string,
+			flagKey: string,
+			value?: PatchComment | InstructionPatch,
+		) => Promise<FeatureFlag | Error>,
+		projectKey: string,
+		flagKey: string,
+		value?: PatchComment | InstructionPatch,
+	): Promise<FeatureFlag>;
+	allFlags(): Promise<FlagConfiguration[] | LDFeatureStoreKindData>;
+	variationDetail(flag: string, context: LDContext): Promise<LDEvaluationDetail>;
+	// Add other methods as needed...
+}
+
+export interface ILaunchDarklyTreeViewProvider extends TreeDataProvider<IFlagTree | IFlagTree[]> {
+	flagNodes: Array<IFlagTree> | null;
+	start(): void;
+	treeLoader(): void;
+	registerCommands(): void;
+	refresh(): void;
+	stop(): void;
+	// Add other methods as needed...
+}
+
+export interface IFlagTree {
+	children: unknown;
+	command?: unknown;
+	flagKey?: string;
+	flagVersion?: number;
+}
+
+export interface ILaunchDarklyAuthenticationSession extends AuthenticationSession {
```

Comment:
IIRC on our gonfalon FE we tend to prefer using types instead of interfaces, though I'm not sure if it makes any difference. 

For example you could do something like this:
```ts
export type LaunchDarklyAuthenticationSession =  AuthenticationSession & {
	refreshToken: string;
	baseUri: string;
	fullUri: string;
	teams: Team[];
	apiToken?: string;
}
```

You might want to ask Alexis for feedback but I'm happy to go with the interface approach.

Reply:
The differences probably don't matter a ton here.

https://www.typescriptlang.org/play#example/types-vs-interfaces